### PR TITLE
fix: error handling when delete non-existent target

### LINKED ISSUE #626: `oci.Storage.Delete()` should return `ErrNotFound` if target is not present in the storage
In the current implementation, a `*PathError` will be return if `target` is not present in the storage, which is not user-friendly.

https://github.com/oras-project/oras-go/blob/0f1dc30a0fd713f3bafcf3bd52693b2e8a8a6fa5/content/oci/storage.go#L109-L117

### PR DESCRIPTION
**Pull request by wangxiaoxuan273**
resolves #626

### CODE REVIEW

File: content/oci/storage.go [hunk 1]
```diff
@@ -108,6 +108,13 @@ func (s *Storage) Push(_ context.Context, expected ocispec.Descriptor, content i
 
 // Delete removes the target from the system.
 func (s *Storage) Delete(ctx context.Context, target ocispec.Descriptor) error {
+	exists, err := s.Exists(ctx, target)
+	if err != nil {
+		return err
+	}
+	if !exists {
+		return fmt.Errorf("target to delete is not present in the storage: %w", errdef.ErrNotFound)
+	}
```
Comment (shizhMSFT) [inline content/oci/storage.go:117]: It introduces a race condition of [TOCTOU](https://en.wikipedia.org/wiki/Time-of-check_to_time-of-use).

Can you check the error returned by `os.Remove()` and return `ErrNotFound` accordingly?

File: content/oci/storage.go [hunk 2]
```diff
@@ -113,7 +114,14 @@ func (s *Storage) Delete(ctx context.Context, target ocispec.Descriptor) error {
 		return fmt.Errorf("%s: %s: %w", target.Digest, target.MediaType, errdef.ErrInvalidDigest)
 	}
 	targetPath := filepath.Join(s.root, path)
-	return os.Remove(targetPath)
+	err = os.Remove(targetPath)
+	if err != nil {
+		if errors.Is(err, fs.ErrNotExist) {
+			return fmt.Errorf("target to delete is not present in the storage: %w", errdef.ErrNotFound)
```
Comment (shizhMSFT) [inline content/oci/storage.go:120]: Can you follow the same error message pattern as line 114?

Comment (wangxiaoxuan273) [reply content/oci/storage.go]: changed accordingly,

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Review by Wwwsylvia — APPROVED** [review]
LGTM

**Comment by Wwwsylvia** [comment]
Sorry, accidentally clicked "close", thought it was "merge"...
image

**Pull request merged**